cookbooks/aws-parallelcluster-config/templates/default/init/fetch_and_run.erb

```diff
@@ -25,11 +26,11 @@ function download_run (){
     tmpfile=$(mktemp)
     trap "/bin/rm -f $tmpfile" RETURN
     if [ "${scheme}" == "s3" ]; then
-      <%= node['cluster']['cookbook_virtualenv_path'] %>/bin/aws --region ${cfn_region} s3 cp ${url} - > $tmpfile || return 1
+      <%= node['cluster']['cookbook_virtualenv_path'] %>/bin/aws --region ${cfn_region} s3 cp ${url} - > $tmpfile || error_exit "Failed to run ${ACTION}, failed to download ${url} using aws s3, return code: $?"
     else
-      wget -qO- ${url} > $tmpfile || return 1
+      wget -qO- ${url} > $tmpfile || error_exit "Failed to run ${ACTION}, failed to download ${url} using wget, return code: $?"
     fi
-    chmod +x $tmpfile || return 1
+    chmod +x $tmpfile || error_exit "Failed to run ${ACTION}, failed to run chmod +x on temporary file, return code: $?"
```

Comment:
minor: the message temporary file doesn't looks clear to me. Can we change it to: failed to make file $tmpfile executable

Reply:
ah I see... what about failed to make the file executable